web_client/analysis-modules/feature-matching.jsx

```diff
@@ -117,7 +117,115 @@ const computeMatch = (data, page) => {
   );
 };
 
-const applyMatch = (data, page) => { /* TODO(opadron): */ };
+const applyMatch = (data, page) => {
+  const truncatePromise = Promise.all([
+    D(actions.truncateAnalysisPages(1, {
+      clear: true,
+      disable: true,
+      remove: false
+    })),
+
+    (
+      applyResultElements
+        ? D(actions.removeAnalysisElement(applyResultElements))
```

Comment:
I promised myself I wasn't going to work on my day off, but I just wanted to say that, looking at your example, I was surprised to have found it quite reasonable.  I will be sure to apply this idea to the relevant parts of the code. Thanks!